### PR TITLE
Init Scopes, Routers and Functions

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: mule-sdk/v/4.0/1_structure/8_scopes.adoc [hunk 1]
```diff
@@ -1,2 +1,158 @@
 [[_scopes]]
-= Scopes
+= Creating Scopes with Mule SDK
+[[scopes]][[scope]]
+:keywords: mule, sdk, scopes, scope
+
+== Introduction
```
Comment (juandesi) [inline mule-sdk/v/4.0/1_structure/8_scopes.adoc]: remove `== Introduction`

File: mule-sdk/v/4.0/1_structure/8_scopes.adoc [hunk 2]
```diff
@@ -1,2 +1,158 @@
 [[_scopes]]
-= Scopes
+= Creating Scopes with Mule SDK
+[[scopes]][[scope]]
```
Comment (juandesi) [inline mule-sdk/v/4.0/1_structure/8_scopes.adoc]: ?

Comment (juandesi) [reply mule-sdk/v/4.0/1_structure/8_scopes.adoc]: pltd

File: mule-sdk/v/4.0/1.6_functions.adoc [hunk 3]
```diff
@@ -1,2 +1,56 @@
 [[_functions]]
-= Functions
+= Contributing Functions with Mule SDK
+:keywords: mule, sdk, functions, function
+
+DataWave is the main Expression Language in Mule 4, and Mule SDK is also aligned with that view,
```
Comment (pablomusumeci) [inline mule-sdk/v/4.0/1.6_functions.adoc]: DataWeave

File: mule-sdk/v/4.0/1.6_functions.adoc [hunk 4]
```diff
@@ -1,2 +1,56 @@
...
+allowing you to contribute Functions to DataWeave from your Module's code in a similar fashion
+of how you'd code a Mule Operation. Using custom Functions will allow yo to reuse code in a new way,
```
Comment (pablomusumeci) [inline mule-sdk/v/4.0/1.6_functions.adoc]: Link to operations

File: mule-sdk/v/4.0/1.6_functions.adoc [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+.DocsModule.java
+[source,java,linenums]
+----
+@Extension(name = "Docs")
+@ExpressionFunctions(DocsFunction.class)
+public class DocsModule {
+
+}
+----
+
```
Comment (pablomusumeci) [inline mule-sdk/v/4.0/1.6_functions.adoc]: Add that "in the DocsFunction class we define the following method"

File: mule-sdk/v/4.0/1.7_routers.adoc [hunk 6]
```diff
@@ -1,2 +1,97 @@
 [[_routers]]
-= Routers
+= Creating Routers with Mule SDK
+:keywords: mule, sdk, routers, router
+
+Conceptually, Routers are Operations that can receive many executable Routes along with a set of Parameters, handling the execution of at one, all of none of them.
+
+Routes should be used for receiving multiple possible paths of execution, where the Parameters
+of the Route can be used to decide whether or not the contained processors should be executed.
+Stereotypes can be used in the Route declaration to define which are the allowed stereotypes of
```
Comment (pablomusumeci) [inline mule-sdk/v/4.0/1.7_routers.adoc]: Where did you say what a stereotype is?

Comment (alegmarra) [reply mule-sdk/v/4.0/1.7_routers.adoc]: that's a different section

File: mule-sdk/v/4.0/1.8_scopes.adoc [hunk 7]
```diff
@@ -1,2 +1,155 @@
...
+= Creating Scopes with Mule SDK
+:keywords: mule, sdk, scopes, scope
+
+Scopes are pretty similar to <<_operations, Operations>>, but whose execution includes executing other child <<_operations, Operations>>.
+This means that a Scope is basically an Operation that receives one or more arguments that are simple <<_parameters, Parameters>> along with a single Chain component and the CompletionCallback.
+
+
+A Chain represent the child <<_operations, Operations>> declared inside the Scope by the application developer and provide an abstraction to execute them.
+
+== Getting Started
```
[the remark this answers is not in the document]
Comment (alegmarra) [reply mule-sdk/v/4.0/1.8_scopes.adoc]: not really

File: mule-sdk/v/4.0/1.7_routers.adoc (hunk 6 above)
Comment (juandesi) [reply mule-sdk/v/4.0/1.7_routers.adoc]: They are not in the MVP, we should modify this or explain them

File: mule-sdk/v/4.0/1.8_scopes.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (juandesi) [reply mule-sdk/v/4.0/1.8_scopes.adoc]: Declaring a Scope

**Pull request merged**